### ISSUE TITLE
Issue with closing windows when the reopenability of the window needs to be considered.

### ISSUE THREAD

**Issue by npbamber** [vassalengine/vassal#11597]
**Overview:** I believe VASSAL adopts the following reasoning "If the user closes a window, which they cannot reopen, then they must want to close the whole game." However the test that VASSAL is using is incorrect. It is only looking at whether there is a show/hide button in the toolbar and not checking the presence of show/hide hotkeys (specifically show or toggle hotkeys). 

**Background:** I am working on a module, which exemplifies the issue.  I have checked against versions 3.6.5 to the latest 3.6.8 build. So this module has 2-5 player sides each of which needs a Hand. There is no theming for the sides, so they are just called Player 1 etc. 
I did not want to clutter up the tool bar with 5 show/hide buttons most of which would be disabled at any given time. Player 2 is was my first approach. It has a player hand. If you are Player 2 then the Toggle Player Hand will hide or show it. Player 1 is just the same as Player 2 but uses a private window rather than a player hand. Player 3 is like Player 2 but also has it's own show/hide window button on the toolbar. I have not bothered with Players 4 and 5 yet. All of these work as I expected with one exception: If I click on the top right hand "X" button of the Player Hand 3 window it hides (as I would like), but for Players 1 and 2 the game is closed (which I don't like). 

The module is here https://www.dropbox.com/s/vume6f9p575cqe7/Turncoats_v0.1-player_hand.vmod?dl=0 

I tried raising the issue on discord starting here: https://discord.com/channels/740127632586178572/911006365626036294/998164552434843738

**Comment by npbamber** [comment]
I have perused the code and identified that I believe the issue is in https://github.com/vassalengine/vassal/blob/master/vassal-app/src/main/java/VASSAL/build/module/Map.java .
```
`@Override
              public void windowClosing(WindowEvent e) {
                if (useLaunchButtonEdit) {
                  topWindow.setVisible(false);
                }
                else {
                  g.getGameState().setup(false);
                }
              }`
```

It is overriding the windowClosing event and makes a decision based on useLaunchButtonEdit alone.
I think it also needs to look at the show and toggle hotkey configs or something.

Also I noticed that the show and hide hotkeys will not show up in the list of hot keys. 

I have updated the test module to check these claims.
I added a Player Hand 4 which uses a simple Window rather than a private window.

I added Alt-[1234] and Ctrl-Alt-[1234] hotkeys . Both claims checked out.
